### PR TITLE
Comment post support:  api & cli

### PR DESCRIPTION
**Pull request by nain-F49FF806**
Greetings, 

This PR completes comments support by adding support for posting comments. 
In addition to the above, post paste has been updated to use defined paste type. 

The typing should help give more confidence in future updates to the code.
Also, we are able to focus sensitive/error-prone tasks like generating cipher parameters in one place. 

Best regards.

### CODE REVIEW

File: src/privatebin.rs [hunk 1]
```diff
@@ -126,6 +164,13 @@ pub type DecryptedCommentsMap = HashMap<String, DecryptedComment>;
 /// comment.id -> [children comment.id]
 pub type CommentsAdjacencyMap = HashMap<String, Vec<String>>;
 
+#[derive(Deserialize, Debug, Serialize, Clone)]
+pub struct PostCommentResponse {
+    pub id: String,
+    pub status: u32,
+    pub url: String,
```
Comment (Mydayyy) [inline src/privatebin.rs:171]: Would changing this to an URL make sense or does that interfere with uniffi?

File: src/privatebin.rs [hunk 2]
```diff
@@ -30,11 +33,12 @@ pub enum PasteFormat {
...
 
-#[derive(Deserialize, Debug, Serialize)]
+#[skip_serializing_none]
+#[derive(Default, Deserialize, Debug, Serialize)]
 pub struct Paste {
-    pub status: i32,
+    pub status: Option<i32>,
     pub id: String,
-    pub url: String,
+    pub url: Option<String>,
```
Comment (Mydayyy) [inline src/privatebin.rs:41]: Would changing this to an URL make sense or does that interfere with uniffi?

**Review by Mydayyy — COMMENTED** [review]
Thank you for the contribution! Just 2 (or basically one) minor questions. For me atleast it compiled without changes when changing those to be an URL, but I am not sure which impact that would have on uniffi

File: src/privatebin.rs (hunk 2 above)
Comment (nain-F49FF806) [reply src/privatebin.rs:41]: There would be no issue with uniffi. 
But the "url" returned by privatebin responses can be weird:

```json
{"status":0,"id":"ba7bf946a542eee1","url":"\/?ba7bf946a542eee1","deletetoken":"9914b0e934253e2e327f9a6477b99f6fff5d1e0e10d3d3bd08d4d9b6a20ab2ea"}
```

So just need to test if it parses correctly.  
Same with above.

Comment (nain-F49FF806) [reply src/privatebin.rs:41]: Update: We do indeed get error when actually parsing it to url. The url value should be fixed upstream before we try to make use of this property.  

```
Error: Json(Error("invalid value: string \"/?518ad640b320f1a4\", expected relative URL without a base", line: 0, column: 0))
```

Comment (nain-F49FF806) [reply src/privatebin.rs:41]: Have opened an issue upstream regarding this https://github.com/PrivateBin/PrivateBin/issues/1294

Comment (Mydayyy) [reply src/privatebin.rs:41]: Hey, 

in that case it appears there is nothing that can be done from our side, so I will proceed and merge this. Thank you for the contribution!

Best Regards
Mydayyy

**Pull request merged**